### ISSUE TITLE
USB3 xHCI detection false-negative when driver is xhci-pci-renesas

### ISSUE THREAD

**Issue by k-s-0** [canonical/checkbox#2333]
### Bug Description

The usb3/storage-server test fails if any driver other than the `xhci_hcd` is detected, such as when the driver is reported as `xhci-pci-renesas`.

I observed this when certifying a Nvidia GB200 NVL72 node: [see the failed test here](https://certification.canonical.com/hardware/202512-38187/submission/472617/test/113947/result/55847002/).

Here are some links to relevant lines in the codebase, not necessarily exhaustive: [1](https://github.com/canonical/checkbox/blob/630ef1b9e80f70aa43f64a8d8d97c97fea2ecb18/checkbox-support/checkbox_support/udev.py#L109), [2](https://github.com/canonical/checkbox/blob/630ef1b9e80f70aa43f64a8d8d97c97fea2ecb18/providers/base/bin/removable_storage_test.py#L696), [3](https://github.com/canonical/checkbox/blob/630ef1b9e80f70aa43f64a8d8d97c97fea2ecb18/providers/base/bin/removable_storage_test.py#L985), [4](https://github.com/canonical/checkbox/blob/630ef1b9e80f70aa43f64a8d8d97c97fea2ecb18/providers/base/bin/removable_storage_test.py#L512), [5](https://github.com/canonical/checkbox/blob/630ef1b9e80f70aa43f64a8d8d97c97fea2ecb18/providers/base/bin/removable_storage_test.py#L1012)

### Cert-blocker Test Case

- [x] cert-blocker

### To Reproduce

1. Provision a device with which uses a driver other than `xhci_hcd`, contains USB 3.0 port(s), and has USB(s) inserted.
2. Launch any Checkbox job which includes the `usb3/storage-server` test (or another USB test which hits the code paths linked).
3. Verify the test failed and contains the note `Disk does not use xhci_hcd.`.

### Expected Result

Job passes.

### Actual Result

Job fails with note: `Disk does not use xhci_hcd.`.

### Environment

Ubuntu 24.04
checkbox deb, 7.0.0\~dev147\~ubuntu24.04.1
Nvidia GB200 NVL72, during USB3 tests

### Relevant log output

```shell
ubuntu@canonical-gb200-single:~$ lsusb -t
/:  Bus 001.Port 001: Dev 001, Class=root_hub, Driver=xhci-pci-renesas/4p, 480M
    |__ Port 001: Dev 002, If 0, Class=Hub, Driver=hub/7p, 480M
        |__ Port 001: Dev 003, If 0, Class=Communications, Driver=cdc_ether, 480M
        |__ Port 001: Dev 003, If 1, Class=CDC Data, Driver=cdc_ether, 480M
/:  Bus 002.Port 001: Dev 001, Class=root_hub, Driver=xhci-pci-renesas/4p, 5000M
    |__ Port 002: Dev 002, If 0, Class=Mass Storage, Driver=usb-storage, 5000M


usb/storage-server:
--------------------
Removable devices currently mounted:
None
Removable devices currently not mounted:
/dev/sda1
--------------------
Found the following mounted usb partitions:
    /dev/sda1 : /tmp/tmpv6saahu3 : 5000000000 bits/s
--------------------
/dev/sda1 (Total Data Size / iteration: 255.9662 MB):
	[Iteration 0] Average Speed: 34.0147
	Summary:
		Total Data Attempted: 255.9662 MB
		Total Time to write: 7.5252 secs
		Average Write Time: 7.5252 secs
		Average Write Speed: 34.0147 MB/s
		--------------------------------
		Device Detected: SuperSpeed USB
		Disk does not use xhci_hcd.
```
